src/core/plugin/luapi_application.h

```diff
@@ -327,6 +384,263 @@ static int applib_layerAction(lua_State* L) {
+ *            ["fill"] = 0,
+ *            ["tool"] = "STROKE_TOOL_PEN",
+ *            ["lineStyle"] = "default"
+ *        })
+ */
+static int applib_addStroke(lua_State* L) {
+    Plugin* plugin = Plugin::getPluginFromLua(L);
+    Control* ctrl = plugin->getControl();
+    PageRef const& page = ctrl->getCurrentPage();
+    Layer* layer = page->getSelectedLayer();
+    Stroke* stroke = new Stroke();
+
+    // Discard any extra arguments passed in
+    lua_settop(L, 1);
+    luaL_checktype(L, 1, LUA_TTABLE);
+    // Push another reference to the table on top of the stack (so we know
+    // where it is, and this function can work for negative, positive and
+    // pseudo indices
+    // stack now contains: -1 => table
+    lua_pushnil(L);
+    // stack now contains: -1 => nil; -2 => table
+    // Fetch table of X values from the Lua stack
+    lua_getfield(L, 1, "x");  // X coords are now at -1
+    lua_pushnil(L);           // X coords are now at -2
+    if (!lua_istable(L, -2))
+        luaL_error(L, "Missing X-Coordinate table!");
+    std::vector<double> xStream;
+    while (lua_next(L, -2)) {
+        double value = lua_tonumber(L, -1);
+        xStream.push_back(value);
+        lua_pop(L, 1);
+    }
+    lua_pop(L, 1);  // Stack is back to normal
+
+    // Fetch table of Y values form the Lua stack
+    // (same drill as above)
+    lua_getfield(L, 1, "y");
+    lua_pushnil(L);
+    if (!lua_istable(L, -2))
+        luaL_error(L, "Missing Y-Coordinate table!");
+    std::vector<double> yStream;
+    while (lua_next(L, -2)) {
+        double value = lua_tonumber(L, -1);
+        yStream.push_back(value);
+        lua_pop(L, 1);
+    }
+    lua_pop(L, 1);
+
+    // Fetch table of pressure values from the Lua stack
+    // (same drill as above)
+    lua_getfield(L, 1, "pressure");
+    lua_pushnil(L);
+    if (!lua_istable(L, -2))
+        luaL_error(L, "Missing Pressure table!");
+    std::vector<double> pressureStream;
+    while (lua_next(L, -2)) {
+        double value = lua_tonumber(L, -1);
+        pressureStream.push_back(value);
+        lua_pop(L, 1);
+    }
+    lua_pop(L, 1);
+    // Make sure all vectors are the same length.
+    if (xStream.size() != yStream.size()) {
+        luaL_error(L, "X and Y vectors are not equal length!");
+    }
+    if (xStream.size() != pressureStream.size())
+        luaL_error(L, "Pressure vector is not equal length!");
+    // Check and make sure there's enough points (need at least 2)
+    if (xStream.size() < 2) {
+        g_warning("Stroke shorter than four points. Discarding. (Has %ld/2)", xStream.size());
+        return 1;
+    }
+    // Add points to the stroke
+    for (long unsigned int i = 0; i < xStream.size(); i++) {
+        Point myPoint = Point(yStream.at(i), xStream.at(i), pressureStream.at(i));
+        stroke->addPoint(myPoint);
+    }
+    // Finish building the Stroke and apply it to the layer.
+    addStrokeHelper(L, stroke, layer);
+    return 1;
```

Comment:
Return the number of returned values (which appears to be 0).
```suggestion
    return 0;
```